source/util/log.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2021-2023 Simon Symeonidis
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef APP_LOG_H
+#define APP_LOG_H
+
+#include <chrono>
```

Comment:
To avoid pulling in a bunch of includes in the header, one thought is that the backend function could just take the final formatted string (e.g. call `vformat()` beforehand), and then it can just go in the .cpp file since it doesn't need to be templated. Then most of the related members like the mutex don't need to be accessible in the header

Reply:
Fixed this too!  I have to AFK for a bit, but once I'm back, I'll make sure that this works on windows too and tick that.